streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -109,27 +111,36 @@ public void register(StateStore store, StateRestoreCallback stateRestoreCallback
             throw new IllegalArgumentException("Store " + store.name() + " has already been registered.");
 
         // check that the underlying change log topic exist or not
-        if (restoreConsumer.listTopics().containsKey(store.name())) {
-            boolean partitionNotFound = true;
-            for (PartitionInfo partitionInfo : restoreConsumer.partitionsFor(store.name())) {
+        String topic = store.name() + STATE_CHANGELOG_TOPIC_SUFFIX;
```

Comment:
I want to distinguish changelog topics with normal topics inside Kafka, this is mainly for operationability: otherwise users have to make sure their state name does not potentially conflict with any topics in a shared cluster, while it might be OK for them to make sure their state name is different with other state names from other jobs.